solutions/standard/variables.tf

```diff
@@ -126,3 +135,114 @@ variable "skip_en_kms_auth_policy" {
+  default     = []
+}
+
+variable "add_bucket_name_suffix" {
+  type        = bool
+  description = "Add random generated suffix (4 characters long) to the newly provisioned COS bucket name. Only used if not passing existing bucket. set to false if you want full control over bucket naming using the 'cos_bucket_name' variable."
+  default     = true
+}
+
+variable "cos_bucket_access_tags" {
+  type        = list(string)
+  default     = []
+  description = "Optional list of access tags to be added to the COS bucket."
+}
+
+variable "cos_bucket_class" {
+  type        = string
+  default     = "smart"
+  description = "The storage class of the newly provisioned COS bucket. Allowed values are: 'standard', 'vault', 'cold', 'smart' (default value), 'onerate_active'"
+  validation {
+    condition     = contains(["standard", "vault", "cold", "smart", "onerate_active"], var.cos_bucket_class)
+    error_message = "Allowed values for cos_bucket_class are \"standard\", \"vault\",\"cold\", \"smart\", or \"onerate_active\"."
+  }
+}
+
+variable "existing_cos_instance_crn" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The CRN of an existing Cloud Object Storage instance. If not supplied, a new instance will be created."
+}
+
+variable "existing_cos_bucket_name" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The name of an existing bucket inside the existing Cloud Object Storage instance. If not supplied, a new bucket will be created."
+}
+
+variable "skip_cos_kms_auth_policy" {
+  type        = bool
+  description = "Set to true to skip the creation of an IAM authorization policy that permits the COS instance created to read the encryption key from the KMS instance. WARNING: An authorization policy must exist before an encrypted bucket can be created"
+  default     = false
+}
+
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM terraform provider to use to manage COS buckets. (`public`, `private` or `direct`). Ensure to enable virtual routing and forwarding (VRF) in your account if using `private`, and that the terraform runtime has access to the the IBM Cloud private network."
+  type        = string
+  default     = "private"
+  validation {
+    condition     = contains(["public", "private", "direct"], var.management_endpoint_type_for_bucket)
+    error_message = "The specified management_endpoint_type_for_bucket is not a valid selection!"
+  }
+}
+
+variable "existing_activity_tracker_crn" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "(Optional) The CRN of an existing Activity Tracker instance. Used to send COS bucket log data and all object write events to Activity Tracker. Only used if not supplying an existing COS bucket."
+}
+
+variable "cos_endpoint" {
+  type        = string
+  description = "The endpoint url for your cos instance."
```

Comment:
Lets be clear this is only needed if providing an existing COS.
Maybe also link to the COS endpoint doc https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-endpoints